src/site/content/en/blog/building-a-settings-component/index.md

```diff
@@ -0,0 +1,959 @@
+more information. 
+{% endAside %}
+
+## Custom Checkbox
+
+Given the following HTML input element, I'll show you how I customized its
+appearance:
+
+```html
+<input type="checkbox">
+```
+
+There's 3 parts to this element we need to customize:
+1. [Checkbox element](#checkbox-element)
+1. [Associated labels](#checkbox-labels)
+1. [Highlight effect](#checkbox-highlight)
+
+### Checkbox element
+
+```css
+input[type="checkbox"] {
+  inline-size: var(--space-sm);   /* increase width */
+  block-size: var(--space-sm);    /* increase height */
+  outline-offset: 5px;            /* focus style enhancement */
+  accent-color: var(--brand);     /* tint the input */
+  position: relative;             /* prepare for an absolute pseudo element */
+  transform-style: preserve-3d;   /* create a 3d z-space stacking context */
+  margin: 0;
+  cursor: pointer;
+}
+```
+
+Besides the pseudo element `transform-style` and `position` styles, it's mostly
+minor opinionated style stuff from me. I like the cursor to be pointer, I like
+outline offsets, default checkboxes are too tiny, and if `accent-color` is
+[supported](https://drafts.csswg.org/css-ui-4/#widget-accent), bring these
+checkboxes into the brand color scheme.
+
+### Checkbox labels
+
+It's important to provide labels for checkboxes for 2 reasons. The first is to
+represent what the checkbox value is used for, to answer "on or off for what?".
+Second is for UX, web users have become accustomed to interacting with
+checkbox's via their associated labels.
+
+{% Video 
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/7GYIFNjNCBdj13juFO7S.mp4",
+  className="w-screenshot", 
+  autoplay="true", 
+  loop="true", 
+  muted="true" 
+%}
+
+<div class="w-columns">
+{% Compare 'better', 'input' %}
+
+```html
+<input 
+  type="checkbox" 
+  id="text-notifications" 
+  name="text-notifications"
+>
+```
+
+{% endCompare %}
+
+{% Compare 'better', 'label' %}
+
+```html
+<label for="text-notifications">
+  <h3>Text Messages</h3>
+  <small>Get notified about all text messages sent to your device</small>
+</label>
+```
+{% endCompare %}
+</div>
+
+On your label, put a `for` attribute that points to a checkbox by ID: `<label
+for="text-notifications">`. On your checkbox, double up both the name and id to
+ensure it's found with varying tools and tech, like a mouse or screenreader:
+`<input type="checkbox" id="text-notifications" name="text-notifications">`.
+`:hover`, `:active` and more come for free with the connection, increasing the
+ways your form can be interacted with.
+
+### Checkbox highlight
+
+I want to keep my interfaces consistent, and the slider element has a nice
+thumbnail highlight that I'd like to use with the checkbox. The thumbnail was
+able to use `box-shadow` and it's `spread` property to scale a shadow up and
+down, but that effect doesn't work here because our checkboxes are, [and should
+be!](https://twitter.com/argyleink/status/1329230409784291328?s=20), square.
```

Comment:
```suggestion
I want to keep my interfaces consistent, and the slider element has a nice
thumbnail highlight that I'd like to use with the checkbox. The thumbnail was
able to use `box-shadow` and it's `spread` property to scale a shadow up and
down. However, that effect doesn't work here because our checkboxes are, [and should
be](https://twitter.com/argyleink/status/1329230409784291328?s=20), square.
```